tests/unittests/Foundation/NSURLTests.m

```diff
@@ -62,7 +62,7 @@ void testNSURLMethod(SEL selector, NSURL* input, id argument, NSURL* expected) {
                     nil,
                     [[NSURL alloc] initWithString:@"http://www.test.com/home.page/asdf"]);
 
-    testNSURLMethod(@selector(URLByDeletingPathExtension), testURL7, nil, [[NSURL alloc] initWithString:@"http://www.test.com/"]);
+    testNSURLMethod(@selector(URLByDeletingPathExtension), testURL7, nil, [[NSURL alloc] initWithString:@"http://www.test.com/.foo"]);
```

Comment:
yes it is and for the second part, yes it would be good, but not the current scope of the PR